midend/simplifyKey.h

```diff
@@ -121,11 +121,10 @@ class DoSimplifyKey : public Transform {
 };
 
 /**
- * Determines whether a key expression is too complex.
- *
- * Any non-lvalues, except built-in method invocation (isValid),
- * are considered complex expressions.
+ *  This pass considers any non-lvalues, except built-in method invocation (isValid),
```

Comment:
This is incorrect, whether a key is complex or not is determined by the policy.
This pass does not know whether non l-values are complex or not.